### PR TITLE
Handle pending changes to `tifffile` defaults and avoid test warnings

### LINKED ISSUE #5337: warnings from skimage.io module in test suite
## Description

The following test suite warnings were observed for the main branch while working on #5336 

```
skimage/io/tests/test_collection.py::TestImageCollection::test_concatenate
skimage/io/tests/test_collection.py::TestImageCollection::test_concatenate_mismatched_image_shapes
skimage/io/tests/test_collection.py::TestImageCollection::test_getitem
skimage/io/tests/test_collection.py::TestImageCollection::test_multiimage_imagecollection
skimage/io/tests/test_collection.py::TestImageCollection::test_slicing
  /home/lee8rx/my_git/pyir/scikit-image/skimage/io/collection.py:292: ResourceWarning: unclosed file <_io.BufferedReader name='/home/lee8rx/.cache/scikit-image/main/data/brick.png'>
    raise

skimage/io/tests/test_collection.py::TestImageCollection::test_concatenate
skimage/io/tests/test_collection.py::TestImageCollection::test_multiimage_imagecollection
  /home/lee8rx/my_git/pyir/scikit-image/skimage/io/collection.py:292: ResourceWarning: unclosed file <_io.BufferedReader name='/home/lee8rx/.cache/scikit-image/main/data/moon.png'>
    raise

skimage/io/tests/test_collection.py::TestImageCollection::test_concatenate_mismatched_image_shapes
skimage/io/tests/test_collection.py::TestImageCollection::test_getitem
skimage/io/tests/test_collection.py::TestImageCollection::test_slicing
  /home/lee8rx/my_git/pyir/scikit-image/skimage/io/collection.py:292: ResourceWarning: unclosed file <_io.BufferedReader name='/home/lee8rx/.cache/scikit-image/main/data/color.png'>
    raise

skimage/io/tests/test_imageio.py::test_imageio_truncated_jpg
  /home/lee8rx/my_git/pyir/scikit-image/skimage/io/tests/test_imageio.py:41: ResourceWarning: unclosed file <_io.BufferedReader name='/home/lee8rx/.cache/scikit-image/main/data/truncated.jpg'>
    imread(fetch('data/truncated.jpg'))

skimage/io/tests/test_tifffile.py::test_tifffile_kwarg_passthrough
  /home/lee8rx/miniconda3/envs/skimage_dev/lib/python3.8/site-packages/tifffile/tifffile.py:2876: DeprecationWarning: TiffFile: the 'fastij' argument is ignored
    warnings.warn(

skimage/io/tests/test_tifffile.py::test_tifffile_kwarg_passthrough
  /home/lee8rx/miniconda3/envs/skimage_dev/lib/python3.8/site-packages/tifffile/tifffile.py:2876: DeprecationWarning: TiffFile: the 'multifile' argument is ignored
    warnings.warn(

skimage/io/tests/test_tifffile.py::test_tifffile_kwarg_passthrough
  /home/lee8rx/miniconda3/envs/skimage_dev/lib/python3.8/site-packages/tifffile/tifffile.py:2876: DeprecationWarning: TiffFile: the 'multifile_close' argument is ignored
    warnings.warn(
```

### LINKED ISSUE #6558: tifffile deprecation warning
When running the test suite, I see:

```
/test_pil.py::test_imread_separate_channels
  /home/stefan/envs/py310/lib64/python3.10/site-packages/tifffile/tifffile.py:1155: DeprecationWarning: <tifffile.TiffWriter.write> data with shape (3, 16, 8) and dtype 'float64' are stored as RGB with separate component planes. Future versions will store such data as MINISBLACK in separate pages by default, unless the 'photometric' and 'planarconfig' parameters are specified.
    result = tif.write(
```

### LINKED ISSUE #6572: tifffile deprecation warning on test_tifffile.py
## Way to reproduce
```python
run pytest on skimage/io/tests/test_tifffile.py
```
Will show deprecation warning issues
```bash
skimage/io/tests/test_tifffile.py ...................................    [100%]

=============================== warnings summary ===============================
skimage/io/tests/test_tifffile.py::TestSave::test_imsave_roundtrip[False-float32-shape1]
skimage/io/tests/test_tifffile.py::TestSave::test_imsave_roundtrip[True-float32-shape1]
  /scikit-image/.venv/lib/python3.10/site-packages/tifffile/tifffile.py:1155: DeprecationWarning: <tifffile.TiffWriter.write> data with shape (10, 10, 3) and dtype 'float32' are stored as RGB with contiguous samples. Future versions will store such data as MINISBLACK in separate pages by default, unless the 'photometric' parameter is specified.
    result = tif.write(

skimage/io/tests/test_tifffile.py::TestSave::test_imsave_roundtrip[False-float32-shape2]
skimage/io/tests/test_tifffile.py::TestSave::test_imsave_roundtrip[True-float32-shape2]
  /scikit-image/.venv/lib/python3.10/site-packages/tifffile/tifffile.py:1155: DeprecationWarning: <tifffile.TiffWriter.write> data with shape (10, 10, 4) and dtype 'float32' are stored as RGB with contiguous samples. Future versions will store such data as MINISBLACK in separate pages by default, unless the 'photometric' parameter is specified.
    result = tif.write(

skimage/io/tests/test_tifffile.py::TestSave::test_imsave_roundtrip[False-int16-shape1]
skimage/io/tests/test_tifffile.py::TestSave::test_imsave_roundtrip[True-int16-shape1]
  /scikit-image/.venv/lib/python3.10/site-packages/tifffile/tifffile.py:1155: DeprecationWarning: <tifffile.TiffWriter.write> data with shape (10, 10, 3) and dtype 'int16' are stored as RGB with contiguous samples. Future versions will store such data as MINISBLACK in separate pages by default, unless the 'photometric' parameter is specified.
    result = tif.write(

skimage/io/tests/test_tifffile.py::TestSave::test_imsave_roundtrip[False-int16-shape2]
skimage/io/tests/test_tifffile.py::TestSave::test_imsave_roundtrip[True-int16-shape2]
  /scikit-image/.venv/lib/python3.10/site-packages/tifffile/tifffile.py:1155: DeprecationWarning: <tifffile.TiffWriter.write> data with shape (10, 10, 4) and dtype 'int16' are stored as RGB with contiguous samples. Future versions will store such data as MINISBLACK in separate pages by default, unless the 'photometric' parameter is specified.
    result = tif.write(

skimage/io/tests/test_tifffile.py::TestSave::test_imsave_roundtrip[False-float64-shape1]
skimage/io/tests/test_tifffile.py::TestSave::test_imsave_roundtrip[True-float64-shape1]
  /scikit-image/.venv/lib/python3.10/site-packages/tifffile/tifffile.py:1155: DeprecationWarning: <tifffile.TiffWriter.write> data with shape (10, 10, 3) and dtype 'float64' are stored as RGB with contiguous samples. Future versions will store such data as MINISBLACK in separate pages by default, unless the 'photometric' parameter is specified.
    result = tif.write(

skimage/io/tests/test_tifffile.py::TestSave::test_imsave_roundtrip[False-float64-shape2]
skimage/io/tests/test_tifffile.py::TestSave::test_imsave_roundtrip[True-float64-shape2]
  /scikit-image/.venv/lib/python3.10/site-packages/tifffile/tifffile.py:1155: DeprecationWarning: <tifffile.TiffWriter.write> data with shape (10, 10, 4) and dtype 'float64' are stored as RGB with contiguous samples. Future versions will store such data as MINISBLACK in separate pages by default, unless the 'photometric' parameter is specified.
    result = tif.write(

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html

```

### PR DESCRIPTION
**Pull request by grlee77**
## Description

closes #5337 
closes #6558
closes #6572

I see no warnings from the IO module on my system, but it is possible some other versions of dependencies may still warn. Let's see how the tests here go.

We need to decide if we should be adding these kwargs in the plugins for backwards compatibility or if our policy is just to dispatch to whatever the tifffile defaults are when writing unless the user supplies kwargs. In this PR, I chose to automatically set the needed `kwargs` internally to the plugin for backwards compatiiblity and for consistent behavior across current and future tifffile versions.

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
  `doc/release/release_dev.rst`.
- There is a bot to help automate backporting a PR to an older branch. For
  example, to backport to v0.19.x after merging, add the following in a PR
  comment: `@meeseeksdev backport to v0.19.x`
- To run benchmarks on a PR, add the `run-benchmark` label. To rerun, the label
  can be removed and then added again. The benchmark output can be checked in
  the "Actions" tab.

### CODE REVIEW

File: skimage/io/_plugins/tifffile_plugin.py [hunk 1]
```diff
@@ -1,6 +1,51 @@
...
 
-from tifffile import imwrite as imsave, imread as tifffile_imread
+
+def imsave(fname, arr, **kwargs):
+    """Load a tiff image to file.
+
+    Parameters
+    ----------
+    fname : str or file
+        File name or file-like-object.
```
Comment (mkcor) [inline skimage/io/_plugins/tifffile_plugin.py:13]:
```suggestion
        File name or file-like object.
```

File: skimage/io/_plugins/tifffile_plugin.py [hunk 2]
```diff
@@ -1,6 +1,51 @@
...
+    arr : ndarray
+        The array to write
```
Comment (mkcor) [inline skimage/io/_plugins/tifffile_plugin.py:15]:
```suggestion
        The array to write.
```

File: skimage/io/_plugins/tifffile_plugin.py [hunk 3]
```diff
@@ -1,6 +1,51 @@
...
+    kwargs : keyword pairs, optional
+        Additional keyword arguments to pass through (see ``tifffile``'s
+        ``imwrite`` function).
+
+    Notes
+    -----
+    Provided by the tifffile library [1]_, and supports many
+    advanced image types including multi-page and floating point.
```
Comment (mkcor) [inline skimage/io/_plugins/tifffile_plugin.py:23]:
```suggestion
    advanced image types including multi-page and floating-point.
```

File: skimage/io/_plugins/tifffile_plugin.py [hunk 4]
```diff
@@ -1,6 +1,51 @@
...
+
+    This implementation will set `photomotric='RGB'` when writing if the first
+    or last axis of arr has shape 3 or 4. To override this, explicitly
```
Comment (mkcor) [inline skimage/io/_plugins/tifffile_plugin.py:26]:
```suggestion
    or last axis of arr has length 3 or 4. To override this, explicitly
```
array has a shape but axis has a length (or size)

File: skimage/io/_plugins/tifffile_plugin.py [hunk 5]
```diff
@@ -1,6 +1,51 @@
...
+    specify the photometric kwarg.
+
+    This implementation will set `planarconfig='SEPARATE'` when writing if the
+    first axis of arr has shape 3 or 4. To override this, explicitly
```
Comment (mkcor) [inline skimage/io/_plugins/tifffile_plugin.py:30]:
```suggestion
    first axis of arr has length 3 or 4. To override this, explicitly
```

File: skimage/io/_plugins/tifffile_plugin.py [hunk 6]
```diff
@@ -1,6 +1,51 @@
...
+        ``imwrite`` function).
+
+    Notes
+    -----
+    Provided by the tifffile library [1]_, and supports many
+    advanced image types including multi-page and floating point.
+
+    This implementation will set `photomotric='RGB'` when writing if the first
+    or last axis of arr has shape 3 or 4. To override this, explicitly
+    specify the photometric kwarg.
```
Comment (mkcor) [inline skimage/io/_plugins/tifffile_plugin.py:27]:
```suggestion
    pass the photometric kwarg.
```
either 'specify' or 'explicitly pass' (right?)

File: skimage/io/tests/test_pil.py [hunk 7]
```diff
@@ -60,14 +59,22 @@ def test_imread_as_gray():
     assert np.sctype2char(img.dtype) in np.typecodes['AllInteger']
 
 
-def test_imread_separate_channels():
+@pytest.mark.parametrize('explicit_kwargs', [False, True])
+def test_imread_separate_channels(explicit_kwargs):
     # Test that imread returns RGBA values contiguously even when they are
```
Comment (mkcor) [inline skimage/io/tests/test_pil.py:64]:
```suggestion
    # Test that imread returns RGB(A) values contiguously even when they are
```

File: skimage/io/tests/test_pil.py [hunk 8]
```diff
@@ -60,14 +59,22 @@ def test_imread_as_gray():
...
     # stored in separate planes.
     x = np.random.rand(3, 16, 8)
     f = NamedTemporaryFile(suffix='.tif')
     fname = f.name
     f.close()
-    imsave(fname, x)
+    # Tifffile is used as backend whenever suffix is .tif or .tiff
+    # To avoid pending changes to tifffile defaults, we must specify this is an
+    # RGB image with separate planes (i.e. channel_axis=0).
```
Comment (mkcor) [inline skimage/io/tests/test_pil.py:72]:
```suggestion
    # RGB image with separate planes (i.e., channel_axis=0).
```

**Review by mkcor — COMMENTED** [review]
Thanks for taking care of this! :pray: 
I do see the warnings from the IO module on my system:
```sh
Python 3.9.4 | packaged by conda-forge | (default, May 10 2021, 22:13:33) 
scikit-image.git@29770dbc9fd34131d69b6f9c2fbb1ead835562cd#egg=scikit_image
tifffile==2022.5.4
```

File: skimage/io/_plugins/tifffile_plugin.py [hunk 9]
```diff
@@ -1,6 +1,51 @@
...
+    kwargs : keyword pairs, optional
+        Additional keyword arguments to pass through (see ``tifffile``'s
+        ``imwrite`` function).
+
+    Notes
+    -----
+    Provided by the tifffile library [1]_, and supports many
+    advanced image types including multi-page and floating-point.
+
+    This implementation will set `photomotric='RGB'` when writing if the first
```
Comment (stefanv) [inline skimage/io/_plugins/tifffile_plugin.py:25]:
```suggestion
    This implementation will set ``photometric='RGB'`` when writing if the first
```

File: skimage/io/_plugins/tifffile_plugin.py [hunk 10]
```diff
@@ -1,6 +1,51 @@
...
+    or last axis of arr has length 3 or 4. To override this, explicitly
+    pass the photometric kwarg.
+
+    This implementation will set `planarconfig='SEPARATE'` when writing if the
```
Comment (stefanv) [inline skimage/io/_plugins/tifffile_plugin.py:29]:
```suggestion
    This implementation will set ``planarconfig='SEPARATE'`` when writing if the
```

**Review by stefanv — COMMENTED** [review]
Looks good to me; just the one typo in the docstring, all the other suggestions are stylistic.

**Review by stefanv — APPROVED** [review]

**Review by mkcor — APPROVED** [review]

**Pull request merged**